### PR TITLE
[FIX] mrp_subcontracting_dropshipping: technical fields

### PR DESCRIPTION
**Pull request by wtaferner**
...should not trigger UserError

**Description of the issue/feature this PR addresses:**
As a display line does not need to fulfill the dropship check, we skip those and we do give the user a hint which lines are potentially not fulfilling the check.

**Current behavior before PR:**
Showstopper for the user if
```python 
display_type = 'line_section' or 'line_note'
```
and no information which line triggered the blocker

**Desired behavior after PR is merged:**
Happy customer or wife, happy life 😊  

Info: @wt-io-it

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by wtaferner** [comment]
@ajf-odoo @amoyaux 
I think you two were author and reviewer, so I guess I ping you here to get this merged.

**Comment by amoyaux** [comment]
Hello @wtaferner Okay but can you describe a bit better how to trigger the issue next time.
The only way in standard I found to recreate was to create a new rule in dropship subcontractor on order with a propagate method set to `Fixed´. So it doesn't found the production in _get_destination_location from the `_prepare_picking` (if it comes from `_prepare_stock_move_vals` then line with display type are already removed). I let you edit the commit message with a use case step by step to reproduce it.

**Comment by wtaferner** [comment]
@amoyaux 
Done, I hope you do not doubt about the issue and how it works or not, but if so, hope my description in the commit message helps to reproduce. Thank you.

**Comment by wtaferner** [comment]
@amoyaux 
Reminder as it is only a commit message change as requested IMO

**Comment by wtaferner** [comment]
@amoyaux 
Still alive? A little line to merge this would be much appreciated and will probably make quite some users at least not unhappy 😉

**Comment by amoyaux** [comment]
Okay just missing .pot for translation

File: addons/mrp_subcontracting_dropshipping/models/purchase.py [hunk 1]
```diff
@@ -21,14 +21,17 @@ def _get_destination_location(self):
...
                 in_bom_products = True
-            else:
-                not_in_bom_products = True
+            elif not order_line.display_type:
+                not_in_bom_products |= order_line
         if in_bom_products and not_in_bom_products:
-            raise UserError(_("It appears some components in this RFQ are not meant for subcontracting. Please create a separate order for these."))
+            raise UserError(
+                _("It appears some components in this RFQ are not meant for subcontracting. Please create a separate order for these.") +
+                '\n\n' + '\n'.join(not_in_bom_products.mapped('name'))
```
Comment (amoyaux) [inline addons/mrp_subcontracting_dropshipping/models/purchase.py:33]: display_name instead of name

Comment (wtaferner) [reply addons/mrp_subcontracting_dropshipping/models/purchase.py:33]: for the `purchase.order.line`?

**Comment by wtaferner** [comment]
@amoyaux 
Still missing an answer and it would be nice if we could merge this as it helps to identify the problematic lines instead of blocking and harassing the user with a useless message in case there are more than two or three lines in the purchase order..

**Comment by wtaferner** [comment]
@amoyaux 
Not sure if you ignore or forgot about it, but I guess this is still valuable to merge...

**Comment by amoyaux** [comment]
robodoo r+

**Pull request closed without merging**